pymc_marketing/mmm/base.py

```diff
@@ -78,132 +67,241 @@ def methods(self) -> List[Any]:
         ]
 
     @property
-    def validation_methods(self) -> List[Callable[["BaseMMM", pd.DataFrame], None]]:
-        return [
-            method
-            for method in self.methods
-            if getattr(method, "_tags", {}).get("validation", False)
-        ]
+    def validation_methods(
+        self,
+    ) -> Tuple[
+        List[Callable[["BaseMMM", Union[pd.DataFrame, pd.Series]], None]],
+        List[Callable[["BaseMMM", Union[pd.DataFrame, pd.Series]], None]],
+    ]:
+        """
+        A property that provides validation methods for features ('X') and the target variable ('y').
+
+        This property scans the methods of the object and returns those marked for validation.
+        The methods are marked by having a _tags dictionary attribute, with either 'validation_X' or 'validation_y' set to True.
+        The 'validation_X' tag indicates a method used for validating features, and 'validation_y' indicates a method used for validating the target variable.
+
+        Returns
+        -------
+        tuple of list of Callable[["BaseMMM", pd.DataFrame], None]
+            A tuple where the first element is a list of methods for 'X' validation, and the second element is a list of methods for 'y' validation.
+
+        Example
+        -------
+        >>> self.validation_methods
```

Comment:
This is not very informative. I suggest removing it.

Reply:
I agree, removed.